### PR TITLE
Add SSL support for Remote.Hosting

### LINKED ISSUE #327: Akka.Remote: need to add TLS settings for DotNetty transport
**Is your feature request related to a problem? Please describe.**

Need to add support for [TLS settings in DotNetty](https://getakka.net/articles/remoting/security.html) via `RemoteOptions` or equivalent - also, do we want to start designing the API knowing that we might have additional non-DotNetty transports in the future?

### PR DESCRIPTION
**Pull request by Arkatufus**
Fixes #327

## Changes
* Add SSL options to DotNetty transport

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
> NOTE: this breaks MS.Ext.Configuration binding because the transport list has to be based on an abstract class.

in that case, might be better off not doing this until we actually need additional transports.

**Comment by Arkatufus** [comment]
Updated the PR comment to explain why configuration binding doesn't work

File: src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApproveRemoting.verified.txt [hunk 1]
```diff
@@ -6,13 +6,66 @@
...
+        public bool? EnableSsl { get; set; }
+        public string? HostName { get; set; }
+        public int? Port { get; set; }
+        public string? PublicHostName { get; set; }
+        public int? PublicPort { get; set; }
+        public Akka.Remote.Hosting.SslOptions Ssl { get; set; }
+        public override string TransportId { get; }
+        protected override void Build(System.Text.StringBuilder builder) { }
+    }
+    public sealed class DotNettyUdpTransportOptions : Akka.Remote.Hosting.DotNettyTransportOptions
```
Comment (Aaronontheweb) [inline src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApproveRemoting.verified.txt]: Remove - not supported in Akka.Remote and most likely won't be.

File: src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApproveRemoting.verified.txt [hunk 2]
```diff
@@ -6,13 +6,66 @@
...
+    }
+    public interface IWithPublicHostNameAndPort : Akka.Remote.Hosting.IWithHostNameAndPort
+    {
+        string? PublicHostName { get; set; }
+        int? PublicPort { get; set; }
+    }
+    public class RemoteOptions
     {
         public RemoteOptions() { }
+        public System.Collections.Generic.List<Akka.Remote.Hosting.RemoteTransportOptions> EnabledTransports { get; set; }
```
Comment (Aaronontheweb) [inline src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApproveRemoting.verified.txt]: You know, I'm thinking that even though we could support this - it's probably a footgun for 99.9999999999999% of users. Make super-users use HOCON if they want this feature.

File: src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApproveRemoting.verified.txt [hunk 3]
```diff
@@ -6,13 +6,66 @@
...
         public int? PublicPort { get; set; }
-        public override string ToString() { }
+    }
+    public abstract class RemoteTransportOptions
+    {
+        protected RemoteTransportOptions() { }
+        public abstract string TransportId { get; }
+        protected abstract void Build(System.Text.StringBuilder builder);
+    }
+    public sealed class SslCertificateOptions
```
Comment (Aaronontheweb) [inline src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApproveRemoting.verified.txt:19]: LGTM

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Left some thoughts

**Comment by Arkatufus** [comment]
Remove multi-transport support

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**